spec/lib/gds_api/whitehall_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+RSpec.describe GdsApi::Whitehall do
+  describe "document_list" do
+    let(:whitehall_adapter) { GdsApi::Whitehall.new(Plek.find("whitehall-admin")) }
+    let(:whitehall_export_page_1) { build(:whitehall_export_index, documents: build_list(:whitehall_export_index_document, 100)) }
+    let(:whitehall_export_page_2) { build(:whitehall_export_index, documents: build_list(:whitehall_export_index_document, 10)) }
+
+    before do
+      stub_request(:get, "https://whitehall-admin.test.gov.uk/government/admin/export/document?lead_organisation=123&type=NewsArticle&page_count=100&page_number=1")
+        .to_return(status: 200, body: whitehall_export_page_1.to_json)
+      stub_request(:get, "https://whitehall-admin.test.gov.uk/government/admin/export/document?lead_organisation=123&type=NewsArticle&page_count=100&page_number=2")
+        .to_return(status: 200, body: whitehall_export_page_2.to_json)
+    end
+
+    it "makes a GET request to whitehall for correct number of pages" do
```

Comment:
Minor, should we tweak this to say it allows you to iterate through a list of documents?

Reply:
Updated the name to `iterates through the correct number of pages`